js/components/interface/query/customComponents/queryResultsControlsComponent.js

```diff
@@ -0,0 +1,132 @@
+define(function (require) {
+  
+  require("../query.less");
+  require("../react-simpletabs.less");
+
+  var React = require('react');
+  var GEPPETTO = require('geppetto');
+
+  class QueryResultsControlsComponent extends React.Component {
+    constructor (props) {
+      super(props);
+
+    }
+
+    replaceTokensWithPath (inputStr, path) {
+      return inputStr.replace(/\$ID\$/gi, path);
+    }
+    
+    getActionString (control, path) {
+      var actionStr = '';
+    
+      if (control.actions.length > 0) {
+        for (var i = 0; i < control.actions.length; i++) {
+          actionStr += ((i != 0) ? ";" : "") + this.replaceTokensWithPath(control.actions[i], path);
+        }
+      }
+    
+      return actionStr;
+    }
+    
+    resolveCondition (control, path, negateCondition) {
+      if (negateCondition == undefined) {
+        negateCondition = false;
+      }
+    
+      var resolvedConfig = control;
+    
+      if (resolvedConfig.hasOwnProperty('condition')) {
+        // evaluate condition and reassign control depending on results
+        var conditionStr = this.replaceTokensWithPath(control.condition, path);
+        if (eval(conditionStr)) {
+          resolvedConfig = negateCondition ? resolvedConfig.false : resolvedConfig.true;
+        } else {
+          resolvedConfig = negateCondition ? resolvedConfig.true : resolvedConfig.false;
+        }
+      }
+    
+      return resolvedConfig;
+    }
+    
+    render () {
+      // TODO: would be nicer to pass controls and config straight from the parent component rather than assume
+      var config = GEPPETTO.QueryBuilder.state.resultsControlsConfig;
```

Comment:
@ddelpiano same, we should remove GEPPETTO.QueryBuilder referencing from within these components ideally

Reply:
as per my reply above.